tests/Unit/RmStepTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+use Symfony\Component\Filesystem\Filesystem;
+use WordPress\Blueprints\BlueprintException;
+use WordPress\Blueprints\Model\DataClass\RmStep;
+use WordPress\Blueprints\Runner\Step\RmStepRunner;
+use WordPress\Blueprints\Runtime\NativePHPRuntime;
+
+
+beforeEach(function() {
+    $this->runtime = new NativePHPRuntime(sys_get_temp_dir());
+
+    $this->step = new RmStepRunner();
+    $this->step->setRuntime($this->runtime);
+
+    $this->fileSystem = new Filesystem();
+});
+
+it('should remove a directory when an absolute path is provided', function() {
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+    $this->fileSystem->mkdir($resolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $resolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it('should remove a directory when a relative path is provided', function () {
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+    $this->fileSystem->mkdir($resolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $unresolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it ('should remove a directory with a subdirectory', function () {
+    $unresolvedSubDir = "dir/subdir";
+    $resolvedSubDir = $this->runtime->resolvePath($unresolvedSubDir);
+    $this->fileSystem->mkdir($resolvedSubDir);
+
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $unresolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it ('should remove a directory with a file', function () {
+    $unresolvedFile = "dir/file.txt";
+    $resolvedFile = $this->runtime->resolvePath($unresolvedFile);
```

Comment:
I know I asked the opposite earlier, but perhaps these names would be easier to understand after all:

```suggestion
    $relativePath = "dir/file.txt";
    $absolutePath = $this->runtime->resolvePath($relativePath);
```

Reply:
As a general rule I would suggest to move from `resolved` -> `absolute` and `unresolved` -> `relative`. This would also apply to the `resolvePath` method. Maybe `toAbsolutePath` (as in Path util) would work best?

Reply:
We're testing using a `NativePHP` runtime so absolute and resolved are synonymous, but the runtime could eventually be something like a remote Docker container and then the resolved path wouldn't exist on a local computer. `toAbsolutePath` doesn't quite express the intention. But perhaps remote runtimes won't make it to the final library release? Let's leave that method name as it is for now, but adjust the variable names in the test cases.

Reply:
Done. 

Not relevant to this pr, but as food for thought  -> this is an internal method, if a certain name would work best at this point in time, and there are no clear plans that definitely invalidate it. Maybe we should do what is best now, and adjust accordioning when a true need arises?

Reply:
Makes sense to me 👍